packages/app-store/alby/components/AlbyPaymentComponent.tsx

```diff
@@ -0,0 +1,184 @@
+      <PaymentChecker {...props.paymentPageProps} />
+      {isPaying && <Spinner className="mt-12 h-8 w-8" />}
+      {!isPaying && (
+        <>
+          {!showQRCode && (
+            <div className="flex gap-4">
+              <Button color="secondary" onClick={() => setShowQRCode(true)}>
+                Show QR
+              </Button>
+              {window.webln && (
+                <Button
+                  onClick={async () => {
+                    try {
+                      if (!window.webln) {
+                        throw new Error("webln not found");
+                      }
+                      setPaying(true);
+                      await window.webln.enable();
+                      window.webln.sendPayment(paymentRequest);
+                    } catch (error) {
+                      setPaying(false);
+                      alert((error as Error).message);
+                    }
+                  }}>
+                  Pay Now
+                </Button>
+              )}
+            </div>
+          )}
+          {showQRCode && (
+            <>
+              <div className="flex items-center justify-center gap-2">
+                <p className="text-xs">Waiting for payment...</p>
+                <Spinner className="h-4 w-4" />
+              </div>
+              <p className="text-sm">Click or scan the invoice below to pay</p>
+              <Link
+                href={`lightning:${paymentRequest}`}
+                className="inline-flex items-center justify-center rounded-2xl rounded-md border border-transparent p-2
+                font-medium text-black shadow-sm hover:brightness-95 focus:outline-none focus:ring-offset-2">
+                <QRCode size={128} value={paymentRequest} />
+              </Link>
+
+              <Button
+                size="sm"
+                color="secondary"
+                onClick={() => copyToClipboard(paymentRequest)}
+                className="text-subtle rounded-md"
+                StartIcon={isCopied ? ClipboardCheck : Clipboard}>
+                Copy Invoice
+              </Button>
+              <Link target="_blank" href="https://getalby.com" className="link mt-4 text-sm underline">
+                Don&apos;t have a lightning wallet?
+              </Link>
+            </>
+          )}
+        </>
+      )}
+      <Link target="_blank" href="https://getalby.com">
+        <div className="mt-4 flex items-center text-sm">
+          Powered by
+          <img title="Alby" src="/app-store/alby/icon.svg" alt="Alby" className="h-8 w-8" />
+          Alby
+        </div>
+      </Link>
+    </div>
+  );
+};
+
+type PaymentCheckerProps = PaymentPageProps;
+
+function PaymentChecker(props: PaymentCheckerProps) {
+  // This effect checks if the booking status has changed to "ACCEPTED"
+  // then reload the page to show the new payment status
+  // FIXME: subscribe to the exact payment instead of polling bookings
```

Comment:
We can change this for a TODO: probably enabling payments for trpc and then you actually listen for any success there.

Reply:
updated, thanks